### ISSUE TITLE
Nested lists support

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-list#13]
It's mostly ready but requires couple of fixes.

The UX aspect of the feature is described in https://github.com/ckeditor/ckeditor5-list/issues/5.

**Comment by scofalik** [comment]
It's hard to say how many fixes it requires and how time consuming they will be, however it's true that it's been thought through and there is code in converters that is supposed to handle intending/outdenting.

**Comment by Reinmar** [comment]
Some cases which you'll need to think on:

1. Wrapping a selection spanning multiple levels of items with a quote.
2. Handling empty list item with nested list (selection isn't rendered there correctly – may be an issue with the view or something).
![feb-19-2017 16-35-36](https://cloud.githubusercontent.com/assets/156149/23103811/79da8f58-f6c1-11e6-9de5-f684c3888d0e.gif)

**Comment by scofalik** [comment]
1. Weeelllll, yeah. This is an issue, because you split the list with blockquote but don't fix indent property. I guess we finally found out a drawback of our implementation.
2. This is weird, actually, it should be covered by `ViewListItemElement`: https://github.com/ckeditor/ckeditor5-list/blob/master/src/viewlistitemelement.js#L40

**Comment by Reinmar** [comment]
> Weeelllll, yeah. This is an issue, because you split the list with blockquote but don't fix indent property. I guess we finally found out a drawback of our implementation.

Am I right that the list feature could post-fix this on change? I think that predicted this.

**Comment by scofalik** [comment]
AFAIR there is no such fixer yet. The problematic part is that each change event triggers model conversion dispatcher. So we can't fix it before model conversion dispatcher gets to the data.

OTOH, the problematic part is only moving. That means that, possibly, we could have this logic in move converter for list.

**Comment by Reinmar** [comment]
> he problematic part is that each change event triggers model conversion dispatcher. So we can't fix it before model conversion dispatcher gets to the data. Or rather, we can, but it could produce some weird results anyway.

I wrote a quick proposal to this: https://github.com/ckeditor/ckeditor5-engine/issues/829 🖌

**Comment by scofalik** [comment]
I went through what we already have in `converters.js` and it looks like nested lists are supported in converters. What I mean is that I haven't found any black hole where given part of code is written with flat list items in mind.

This means that we should just start fixing reported bugs. I already reported two bugs I found when testing nested list, I also reported one of the bugs @Reinmar posted here.

For now, I'll close this issue and as I said, let's focus on bugs.
